limiter/lifo_blocking.go

```diff
@@ -252,6 +300,12 @@ func (l *LifoBlockingLimiter) tryAcquire(ctx context.Context) core.Listener {
 		// Remove the holder from the backlog.
 		evict()
 		return nil
+	case <-ctxDone:
```

Comment:
If `backlogEvictDoneCtx` is false, `ctxDone` will end up nil and block forever. Preserving existing behavior